sql/core/src/main/scala/org/apache/spark/sql/streaming/ui/StreamingQueryStatisticsPage.scala

```diff
@@ -150,10 +150,10 @@ private[ui] class StreamingQueryStatisticsPage(parent: StreamingQueryTab)
     if (query.lastProgress.eventTime.containsKey("watermark")) {
       val watermarkData = query.recentProgress.flatMap { p =>
         val batchTimestamp = parseProgressTimestamp(p.timestamp)
-        val watermarkValue = parseProgressTimestamp(p.eventTime.get("watermark"))
```

Comment:
when there is not watermark metrics, `p.eventTime.get("watermark")` will return a `null`. NPE will be thrown in `parseProgressTimestamp`.

Reply:
Could you give more details about the NPE? We already check the `watermark` key in lastProgress.

Reply:
Is there a case earlier batch doesn't have "watermark" while later batch does? It'd be nice if you could refer somewhere you encounter this.

Reply:
https://github.com/apache/spark/blob/b665d5881915f042930f502bcc3c6ee3cb00c50d/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala#L244-L246

It's possible that lastProgress.eventTime contains `watermark` but recentProgress doesn't, when `hasNewData` = true.

Reply:
@HeartSaVioR @xuanyuanking Ignore this false alarm